docs_src/design/adr/devops/0010-Release-Artifacts.md

```diff
@@ -0,0 +1,95 @@
+# Release Artifacts
+
+## Status
+
+In Review 06/10/2020
+
+## Context
+
+During the Geneva release of EdgeX Foundry the DevOps WG transformed the CI/CD process with new Jenkins pipeline functionality. After this new functionality was added we also started adding release automation. This new automation is outlined in ADR 0007 Release Automation. However, in ADR 0007 only three release artifact types are outlined. This document is meant to be a living document to try to outline the artifact types for an EdgeX Foundry release and should be modified as that changes.
+
+## Release Artifact Types
+
+### Docker Images
+
+*Tied to Code Release?* Yes
+
+Docker images are released for every named release of EdgeX Foundry. During development the community releases images to the `docker.staging` repository in [Nexus](http://nexus3.edgexfoundry.org). At the time of release we promote the last tested image from `docker.staging` to `docker.release`. In addition to that we will publish the docker image on [DockerHub](https://hub.docker.com/orgs/edgexfoundry).
+
+#### Nexus Retention Policy
+
+##### docker.snapshots
+Retention Policy: 90 days since last download
+
+Contains: Docker images that are not expected to be released. This will contain EdgeX that images are built with each merge and should be available to test things against a specific commit. In addition to this we also build images to optimize the builds in the CI infrastructure.
+
+Docker Tags Used: Git commit, Version-Git commit, Version (ci-build-images only), Latest (ci-build-images-only)
+
+##### docker.staging
+Retention Policy: 180 days since last download
```

Comment:
Any reason we're keeping these for longer than the snapshots? 6 months seems like a long time to be keeping release candidates around...

Reply:
The LF uses staging as the holding repo before docker images are pushed to release or DockerHub

Reply:
100 days is more like 30 months .. but from last download -- is that a user download of the image, this re-setting some freshness flag? Would this open up an attack vector ... periodically try to download an image to eventually run out of disk space?

Reply:
This is the repository that we would use to store potential release candidates really the time is just to take of the case of images that get less action. IE: Somebody updates the code a month into the release cycle and it could be deleted before release 4-5 months later.

The Linux Foundation hasn't really brought up the problem of disk space with us.. but in general the images in Nexus don't get much action towards them except for the images used by the testing frameworks.

Reply:
I was just asking the question because 6 months seemed like a long time. I'm fine with this as is then...